### PR TITLE
Add example IPython notebook for 2+1 dim Darcy flow problem

### PR DESCRIPTION
**Pull request by m4e7**
Migrate the example from [`fourier_2d.py`](https://github.com/neuraloperator/neuraloperator/blob/master/fourier_2d.py) from the old `master` branch to [`main:examples`](https://github.com/neuraloperator/neuraloperator/tree/main/examples) as `fourier_2d.ipynb` [1]. This was able to use the preexisting `darcy_(train|test)_\d+.pt` files without need for reformatting.

Additionally fix "default arg is mutable" warning in several methods.

[1] TODO

NOTE: this PR is based on top of https://github.com/neuraloperator/neuraloperator/pull/186. It will be rebased on [neuraloperator:main](https://github.com/neuraloperator/neuraloperator/tree/main) when the former PR is merged.

### CODE REVIEW

**Comment by m4e7** [comment]
This example doesn't seem to add much beyond what's already in the documented example ["Training a TFNO on Darcy-Flow"](https://neuraloperator.github.io/neuraloperator/dev/auto_examples/plot_FNO_darcy.html).

Furthermore, the trained network seems to exaggerate flow compared to the ground truth from the numerical solver (https://github.com/m4e7/neuraloperator/blob/3b7ad9216063389927771503dbecc182f171ef1c/examples/fourier_2d.ipynb), as seen by the error and error2 plots, where `error2 := (out / 2.0) - y`

**Comment by m4e7** [comment]
The boundary of the learned operator continues to seem to be a problem. Most test outputs from the model predict the boundary to be around -1.0, while the ground truth from the solver seem to stay within [0.0, +1.0]. Increasing the number of epochs and/or the learning rate doesn't seem to improve performance in this aspect. Playing with the `domain_padding_mode` parameter for `FNO()` didn't help either, though `domain_padding_mode=symmetric` did take 4x longer to train per epoch (~13s, up from 3 seconds on RTX).
